var/spack/repos/builtin/packages/libvdwxc/package.py

```diff
@@ -19,15 +19,19 @@ class Libvdwxc(AutotoolsPackage):
     variant("mpi", default=True, description="Enable MPI support")
     variant("pfft", default=False, description="Enable support for PFFT")
 
-    depends_on("fftw@3:", when="~mpi")
+    depends_on("fftw-api@3")
     depends_on("mpi@2:", when="+mpi")
-    depends_on("fftw@3:+mpi", when="+mpi")
     depends_on("pfft", when="+pfft")
 
     # pfft needs MPI
     conflicts("~mpi", "+pfft")
 
     def configure_args(self):
+        if ('+mpi' in self.spec and '^fftw' in self.spec
+                and '^fftw +mpi' not in self.spec):
+            raise InstallError('libvdwxc+mpi requires fftw+mpi when'
+                               ' using fftw as fftw-api provider')
```

Comment:
Does it work if you say something like:
```python
conflicts('^fftw~mpi', when='+mpi')
```
instead? Framed in this way it will be helpful to forbid a part of the search space to the new concretizer.

Reply:
yes, seems to work, thanks a lot!